core/src/streams.rs

```diff
@@ -331,6 +341,152 @@ impl<'gc> NetStream<'gc> {
+    /// downloads.
+    ///
+    /// `offset` is in milliseconds.
+    ///
+    /// This function should be run during stream ticks and *not* called by AVM
+    /// code to service seek requests.
+    pub fn execute_seek(self, context: &mut UpdateContext<'_, 'gc>, offset: f64) {
+        #![allow(clippy::explicit_auto_deref)] //Erroneous lint
+
+        self.trigger_status_event(
+            context,
+            vec![("code", "NetStream.Seek.Notify"), ("level", "status")],
+        );
+
+        // Ensure the container stream type is known before continuing.
+        if self.0.read().stream_type.is_none() && !self.sniff_stream_type(context) {
+            return;
+        }
+
+        let mut write = self.0.write(context.gc_context);
+
+        if let Some(sound) = write.sound_instance {
+            context.stop_sounds_with_handle(sound);
+            context.audio.stop_sound(sound);
+
+            write.sound_instance = None;
+            write.audio_stream = None;
+        }
+
+        if matches!(write.stream_type, Some(NetStreamType::Flv { .. })) {
+            let slice = write.buffer.to_full_slice();
+            let buffer = slice.data();
+            let mut reader = FlvReader::from_parts(&*buffer, write.offset);
+            let skipping_back = write.stream_time > offset;
+
+            loop {
+                if skipping_back {
+                    let res = FlvTag::skip_back(&mut reader);
+                    if matches!(res, Err(FlvError::EndOfData)) {
+                        //At start of video, can't skip further back
+                        break;
+                    }
+
+                    if let Err(e) = res {
+                        tracing::error!("FLV tag parsing failed during seek backward: {}", e);
+                        break;
+                    }
+                }
+
+                let old_position = reader
+                    .stream_position()
+                    .expect("valid stream position when seeking");
+
+                let tag = FlvTag::parse(&mut reader);
+                if matches!(tag, Err(FlvError::EndOfData)) {
+                    //At end of video, can't skip further forward
+                    break;
+                }
+
+                if let Err(e) = tag {
+                    tracing::error!("FLV tag parsing failed during seek forward: {}", e);
+                    break;
+                }
+
+                if skipping_back {
+                    //Tag position won't actually move backwards if we don't do this.
+                    reader
+                        .seek(SeekFrom::Start(old_position))
+                        .expect("valid backseek position");
+                }
+
+                let tag = tag.unwrap();
+                write.stream_time = tag.timestamp as f64;
+
+                if skipping_back && write.stream_time > offset
+                    || !skipping_back && write.stream_time < offset
```

Comment:
Shouldn't this have either a `>=` or a `<=`, or both?
In case of equality here, we can stop, no matter which direction are we skipping, no?

Reply:
No, sorry, scratch that. The comparison is the other way around than how I interpreted it initially, and it's the condition for `continue`, not `break`. Pardon me, it was quite late...